server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/model/assertion/Assertion.java

```diff
@@ -21,5 +21,508 @@
+    final String prefix = "[computeDiscrepancy]";
+    OptionalInt result = OptionalInt.empty();
+
+    LOGGER.debug(String.format("%s Computing discrepancy for CVR ID %d, Assertion ID %d, " +
+        "contest %s.", prefix, cvr.id(), id, contestName));
+
+    // Get CVRContestInfo matching this assertion's contest from the CVR and audited ballot.
+    final Optional<CVRContestInfo> cvrInfo = cvr.contestInfoForContestResult(contestName);
+    final Optional<CVRContestInfo> acvrInfo = auditedCVR.contestInfoForContestResult(contestName);
+
+    if (auditedCVR.recordType() == CastVoteRecord.RecordType.PHANTOM_BALLOT) {
+      // There is no matching ballot for this CVR (the audited ballot is a phantom), and the
+      // relevant contest for this assertion is on the CVR. If the contest has not been recorded
+      // as being on this CVR, the worst-case discrepancy in this instance is a 1-vote
+      // overstatement (we treat the CVR score as being 0 in this case, and if the ACVR score
+      // is -1, we get a 1-vote overstatement).
+      if (cvr.recordType() == CastVoteRecord.RecordType.PHANTOM_RECORD) {
+        LOGGER.debug(String.format("%s Phantom ballot, phantom record.", prefix));
+        result = OptionalInt.of(2);
+      }
+      else {
+        LOGGER.debug(String.format("%s Phantom ballot.", prefix));
+        result = cvrInfo.map(this::computeDiscrepancyPhantomBallot)
+            .orElseGet(() -> OptionalInt.of(1));
+      }
+    }
+    else if (cvr.recordType() == CastVoteRecord.RecordType.PHANTOM_RECORD) {
+      LOGGER.debug(String.format("%s Phantom record.", prefix));
+      // Similar to the phantom ballot, we use the worst case scenario.
+      result = OptionalInt.of(2);
+
+      if (acvrInfo.isPresent() && acvrInfo.get().consensus() == CVRContestInfo.ConsensusValue.YES) {
+        // Compute ballot score for this assertion.
+        final int acvrScore = score(acvrInfo.get());
+
+        // Based on the ballot score, what is the maximum discrepancy we could have for the
+        // phantom CVR and given ballot?
+        result = OptionalInt.of(1 - acvrScore);
+      }
+    }
+    else if (cvrInfo.isPresent() && acvrInfo.isPresent()) {
+      if (acvrInfo.get().consensus() == CVRContestInfo.ConsensusValue.NO) {
+        LOGGER.debug(String.format("%s Lack of consensus, treat as phantom ballot.", prefix));
+        // A lack of consensus for this contest between auditors is treated as if the ballot is a
+        // phantom ballot.
+        result = computeDiscrepancyPhantomBallot(cvrInfo.get());
+      }
+      else {
+        // First, determine whether there is a difference in the votes on the CVR vs the ballot.
+        // If there is no difference, there is no discrepancy.
+        final boolean recordsSame = cvrInfo.get().choices().equals(acvrInfo.get().choices());
+        if (recordsSame) {
+          result = OptionalInt.empty();
```

Comment:
I think this is unncecessary, since you've set result=OptionalInt.empty(); initially anyway. That is, I think ln 373-385 could all be one else if (!cvrInfo.get().choices().equals ... ) {
   optionalint.of(cvrscore - acvrscrore)
}